main.go

```diff
@@ -301,9 +301,16 @@ func readResponseBody(req *http.Request, resp *http.Response) string {
 
 		if saveOutput == true {
 			// TODO(dfc) handle Content-Disposition: attachment
-			// TODO(dfc) handle the case where someone calls
-			// httpstat -O http://example.com/
 			filename = path.Base(req.URL.RequestURI())
+
+			//Save `http://example.com:4321/` to `example.com`
+			if filename == "/" {
+				host, _, err := net.SplitHostPort(req.URL.Host)
```

Comment:
I prefer this approach, rather than guessing, we just say there is no name to be inferred and they can use -o